### PR TITLE
GUI: Add overlay to show demo status on thumbnails

### PR DESCRIPTION
**Pull request by tag2015**
Following a discussion on discord, this patch reuses most of the already existing code
to add an overlay to show extra data present in the gameinfo. As of now, only the "demo" status
is shown, using a 50% transparent white wording at the bottom of the thumbnail.

### CODE REVIEW

**Comment by lotharsm** [comment]
Hello, thanks a lot for your contribution!

Could you please modify the commit message according to our guidelines[1], e.g. prefix them with the `GUI:` subsystem?

[1] https://wiki.scummvm.org/index.php?title=Commit_Guidelines

**Comment by tag2015** [comment]
> Hello, thanks a lot for your contribution!
> 
> Could you please modify the commit message according to our guidelines[1], e.g. prefix them with the `GUI:` subsystem?
> 
> [1] https://wiki.scummvm.org/index.php?title=Commit_Guidelines

Of course, sorry about that. Should be fine now

File: gui/widgets/grid.cpp [hunk 1]
```diff
@@ -415,6 +425,12 @@ const Graphics::ManagedSurface *GridWidget::platformToSurface(Common::Platform p
 	return _platformIcons[platformCode];
 }
 
+const Graphics::ManagedSurface *GridWidget::demoToSurface(const Common::String extraString) {
+	if (! extraString.contains("Demo") )
```
Comment (bluegr) [inline gui/widgets/grid.cpp:429]:
```suggestion
	if (!extraString.contains("Demo") )
```

**Comment by bluegr** [comment]
Nice! Could you please provide a screenshot of this functionality?

**Comment by tag2015** [comment]
> Nice! Could you please provide a screenshot of this functionality?
Sure, it looks like this
demooverlay

**Comment by bluegr** [comment]
Nice work, thanks!
There are some intermediate debug related commits that are really not needed, plus an unrelated commit for the SCUMM engine. Squashing to keep this as a single commit (since it does fit into one commit).

**Pull request merged**